awx/main/tasks.py

```diff
@@ -429,46 +429,57 @@ def execution_node_health_check(node):
 
 
 def inspect_execution_nodes(instance_list):
-    node_lookup = {}
-    for inst in instance_list:
-        if inst.node_type == 'execution':
-            node_lookup[inst.hostname] = inst
+    with advisory_lock('inspect_execution_nodes_lock', wait=True):
```

Comment:
Added lock to avoid instance group policy change race conditions. I made the scope of the lock much narrower initially, then figured that really the whole operation here should probably be atomic. I went back and forth on this, though, and could see a good case for going either way.